### ISSUE TITLE
Batch verify not working

### ISSUE THREAD

**Issue by saideng** [Electron-Labs/ed25519-circom#66]
Thanks for the repo, and it is very cool!
I wanted to try out batch verify but the tests failed. Does it work? Do you have benchmarks of the batch verify circuit?

**Comment by rahulghangas** [comment]
@saideng the test is seems to be working for me. What's the error you're getting?

A thing to note, the test will require more than 9-10G of memory, does your machine running the test satisfy that?

**Comment by saideng** [comment]
> @saideng the test is seems to be working for me. What's the error you're getting?
> 
> 
> 
> A thing to note, the test will require more than 9-10G of memory, does your machine running the test satisfy that?

The test case with one signature works, but the test cases with two and three signatures do not work for me. I am using MacBook Pro with 16gb memory.

**Comment by jinankjain** [comment]
@saideng By any chance is it M1?

**Comment by rahulghangas** [comment]
@saideng this is likely a resource issue, the test case with 2 and 3 signatures would require more memory (about 20 and 30G). You can add swap space on your mac and use that to compile the circuits and compute corresponding witnesses. If you're getting a specific error, could you post that so that we can investigate further

**Comment by saideng** [comment]
@jinankjain Yes, it is M1. @rahulghangas I tried to compile a 3 signature batch verify and got # of Constraints: 10953937.

I think this is not scalable unless using recursive construction. Have you tried other tools or libraries for recursive SNARK?
